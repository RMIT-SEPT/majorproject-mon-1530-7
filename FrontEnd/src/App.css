.App {
  text-align: center;
  background-color: rgba(164, 214, 197, 0.904);
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.page-container{
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.content-wrap{
  flex:1;
}