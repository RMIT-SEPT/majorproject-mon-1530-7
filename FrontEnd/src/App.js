import React from "react";
import { Redirect } from 'react-router';
import NavigationBar from "./components/layouts/Navbar.js";
import Footer from "./components/layouts/Footer.js";
import { Switch, Route, BrowserRouter } from "react-router-dom";
import LoginPage from "./components/pages/LoginPage";
import HomePage from "./components/pages/HomePage";
import AboutPage from "./components/pages/AboutPage";
import ContactPage from "./components/pages/ContactPage";
import CustomerBookingPage from "./components/pages/CustomerBookingPage.js";
import NewEmployeePage from "./components/pages/NewEmployeePage.js";

import AdminPastBooking from "./components/pages/AdminPastBooking.js";
import CustomerPastBookingsPage from "./components/pages/CustomerPastBookingsPage.js";

import Dashboard from "./components/pages/Dashboard.js";
import AdminDashboard from "./components/pages/AdminDashboard.js";
import Account from "./components/pages/Account.js";
import Profile from "./components/pages/Profile.js";

import ManageEmp from "./components/pages/ManageEmp.js";
import EmpDetails from "./components/pages/EmpDetails.js";

import UserProfile from "./UserProfile.js"
import AdminPastBookingsPage from "./components/pages/AdminPastBookingsPage.js";

function App() {
  console.log(UserProfile.getLoggedIn())
  if(UserProfile.getLoggedIn()) {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
                <Route exact path="/" component={HomePage} />,
                <Redirect from="/login" to="dashboard" />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Route path="/dashboard" component={Dashboard} >
                  {UserProfile.getAdmin() ? <Redirect to="/admin-dashboard" /> : <Dashboard />}
                </Route>,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Route path="/account" component={Account} >
                  {UserProfile.getLoggedIn() ? <Account /> : <Redirect to="/login"/>}
                </Route>,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} /> 
          
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
  else {
    return (
      <div className="page-container">
        <div className="content-wrap">
          <BrowserRouter>
            <NavigationBar loggedIn={UserProfile.getToken()} />
            <Switch>
              
                <Route exact path="/" component={HomePage} />,
                <Route path="/login" component={LoginPage} />,
                <Route path="/about" component={AboutPage} />,
                <Route path="/contact" component={ContactPage} />,
                <Route path="/booking" component={CustomerBookingPage} />,
                <Route
                  path="/customer-past-bookings"
                  component={CustomerPastBookingsPage}
                />,
                <Redirect from="/dashboard" to="/login" />,
                <Route path="/admin-dashboard" component={AdminDashboard} >
                  {UserProfile.getAdmin() ?   <AdminDashboard /> : <Redirect to="/dashboard" /> }
                </Route>,
                <Redirect from="/account" to="/login" />,
                <Route path="/new-employee" component={NewEmployeePage} />,
                <Route path="/profile" component={Profile} />,
                <Route path="/manage-emp" component={ManageEmp} />,
                <Route path="/emp-details" component={EmpDetails} />  
            </Switch>
          </BrowserRouter>
        </div>
        <Footer />
      </div>
    );
  }
}
 

export default App;
