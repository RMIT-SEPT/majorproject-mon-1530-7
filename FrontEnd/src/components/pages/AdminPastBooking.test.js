import React from 'react';
import { MemoryRouter } from 'react-router';
import App from '../../App';
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AdminPastBookingLayout from '../layouts/AdminPastBookingLayout';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing Admin Past Bookings', () => {
    it('Admin Past Booking table components should be present ', () => {
        const component = mount(<MemoryRouter initialEntries={["/admin-past-bookings"]}>
            <App />
        </MemoryRouter>
        );
        expect(component.contains(AdminPastBookingLayout));
    })
})
