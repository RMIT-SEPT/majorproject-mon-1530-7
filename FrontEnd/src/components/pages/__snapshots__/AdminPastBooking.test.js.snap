// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Admin Past Bookings renders correctly 1`] = `
<Jumbotron
  fluid={false}
  id="jumbotron-cus-book-page"
>
  <Container
    fluid={false}
  >
    <h2
      className="h2-cus-book-page"
    >
      Admin Booking History
    </h2>
  </Container>
  <Container
    className="customerBookingPageContainer"
    fluid={false}
  >
    <Card
      body={false}
      border="light"
      className="shadow p-3 mb-5 bg-white rounded"
      style={
        Object {
          "width": "68rem",
        }
      }
    >
      <CardHeader
        as="h6"
      >
        <Row
          noGutters={false}
        >
          <Col>
            Service
          </Col>
          <Col>
            Employee
          </Col>
          <Col>
            Date
          </Col>
          <Col>
            Time
          </Col>
        </Row>
      </CardHeader>
      <CardBody>
        <AdminPastBookingLayout />
      </CardBody>
    </Card>
  </Container>
</Jumbotron>
`;
